### ISSUE TITLE
Incorrect string convert

### ISSUE THREAD

**Issue by Giantaxe** [PomeloFoundation/Pomelo.EntityFrameworkCore.MySql#329]
### Model
```
public class User
{
	public int Id { get; set; }
	public string Name { get; set; }
}

public class MyContext : DbContext
{
	public DbSet<User> Users { get; set; }
	
	protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
		=> optionsBuilder.UseMySql(@"User Id=root;Password=root;Host=localhost;Database=eftest;Persist Security Info=True;Convert Zero Datetime=True;");
}
```

### The issue
The .ToString() function breaks the mysql CONVERT function

### Example
```
using (var db = new MyContext())
{
	// create options for html select
	var options = db.Users.Select(u => new SelectListItem()
	{
		Value = u.Id.ToString(),
		Text = u.Name
	}).ToList();
}
```
Generates the following query
```
SELECT CONVERT(VARCHAR(11), `u`.`Id`) AS `Value`, `u`.`Name` AS `Text`
FROM `users` AS `u`
```

Expected query
```
SELECT CONVERT(`u`.`Id` , CHAR(11)) AS `Value`, `u`.`Name` AS `Text`
FROM `users` AS `u`
```

1. VARCHAR(11) is a invalid conversion type, should be CHAR(11)
2. the fields should be the other way around : 
```
CONVERT(`u`.`Id`, CHAR(11))
```

### Stacktrace
```
Failed executing DbCommand (53ms) [Parameters=[], CommandType='Text', CommandTimeout='0']
SELECT CONVERT(VARCHAR(11), `u`.`Id`) AS `Value`, `u`.`Name` AS `Text`
FROM `users` AS `u`
MySql.Data.MySqlClient.MySqlException (0x80004005): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'VARCHAR(11), `u`.`Id`) AS `Value`, `u`.`Name` AS `Text`
FROM `users` AS `u`' at line 1
   at MySql.Data.MySqlClient.MySqlDataReader.ActivateResultSet(ResultSet resultSet)
   at MySql.Data.MySqlClient.MySqlDataReader.<ReadFirstResultSetAsync>d__62.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at MySql.Data.MySqlClient.MySqlDataReader.<CreateAsync>d__61.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at MySql.Data.MySqlClient.CommandExecutors.TextCommandExecutor.<ExecuteReaderAsync>d__3.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteDbDataReader(CommandBehavior behavior)
   at Microsoft.EntityFrameworkCore.Storage.Internal.MySqlRelationalCommand.<ExecuteAsync>d__3.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.EntityFrameworkCore.Storage.Internal.MySqlRelationalCommand.Execute(IRelationalConnection connection, DbCommandMethod executeMethod, IReadOnlyDictionary`2 parameterValues, Boolean closeConnection)
   at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.ExecuteReader(IRelationalConnection connection, IReadOnlyDictionary`2 parameterValues)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryingEnumerable`1.Enumerator.BufferlessMoveNext(Boolean buffer)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryingEnumerable`1.Enumerator.MoveNext()
   at Microsoft.EntityFrameworkCore.Query.Internal.LinqOperatorProvider.ExceptionInterceptor`1.EnumeratorExceptionInterceptor.MoveNext()
   at System.Collections.Generic.List`1.AddEnumerable(IEnumerable`1 enumerable)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at Test.Program.Main(String[] args) in D:\Sourcecodes\EFTest\Program.cs:line 22
```

### Further technical details

MySQL version: 5.7.17
Operating system: Server 2012
Pomelo.EntityFrameworkCore.MySql version: 2.0.0-preview2-10046

Other details about my project setup: Visual Studio 2017 Preview 4

**Comment by Kagamine** [comment]
@mguinness Could you please take a look at the ToString expression visitor?

**Comment by souchprod** [comment]
Hi, i made a couple of small changes & a test case to fix this one

**Comment by garvincasimir** [comment]
@souchprod  I just encountered this issue in my Mysql tests. Are you planning to merge your fix soon?

**Comment by Kagamine** [comment]
Since #348 covered this issue, close this one.

**Comment by souchprod** [comment]
Hi @garvincasimir, will be merged asap once #346 will be validated and merged. Shouldn't be long.
